components/browser/state/src/main/java/mozilla/components/browser/state/reducer/BrowserStateReducer.kt

```diff
@@ -39,3 +42,38 @@ internal object BrowserStateReducer {
         }
     }
 }
+
+/**
+ * Finds the corresponding tab in the [BrowserState] and replaces it using [update].
+ * @param tabId ID of the tab to change.
+ * @param update Returns a new version of the tab state. Must be the same class,
+ * preferably using [SessionState.copySession].
+ */
+@Suppress("Unchecked_Cast")
+internal fun BrowserState.updateTabState(
+    tabId: String,
+    update: (SessionState) -> SessionState
+): BrowserState {
+    val newTabs = tabs.updateTabs(tabId, update) as List<TabSessionState>?
+    if (newTabs != null) return copy(tabs = newTabs)
```

Comment:
I always prefer explicit blocks with `{}`. But since our tooling does not complain I will accept it. 🙈